workflow/react-project/src/store/randomCocktail/index.jsx

```diff
@@ -1,27 +1,48 @@
 import { createSlice } from '@reduxjs/toolkit';
-import fetchRandomCocktail from './asyncThunks';
+import { fetchRandomCocktail } from './action';
+import { normalizeRandomCocktailData } from './utils';
 
 const initialState = {
-  details: [],
-  status: 'initial',
+  details: {},
+  loading: false,
 };
 
 const randomCocktailSlice = createSlice({
   name: 'randomCocktail',
   initialState,
   reducers: {
     deleteRandomCocktailInformation: state => {
-      state.details = [];
-      state.status = 'initial';
+      state.details = {};
     },
   },
   extraReducers: {
     [fetchRandomCocktail.pending]: state => {
-      state.status = 'pending';
+      state.loading = true;
     },
     [fetchRandomCocktail.fulfilled]: (state, action) => {
-      state.details = action.payload;
-      state.status = 'fulfilled';
+      const cocktailData = normalizeRandomCocktailData(action.payload);
+      state.details.cocktailName = cocktailData?.strDrink;
+      state.details.cocktailPicture = cocktailData?.strDrinkThumb;
+      state.details.instructions = cocktailData?.strInstructions;
+
+      state.details.ingredientsDetails = Object.entries(cocktailData || {}).reduce((acc, [key, value]) => {
+        if (key.includes('strIngredient') && value) {
+          const quantityKey = key.replace('strIngredient', 'strMeasure');
+          const measure = cocktailData[quantityKey];
+
+          return [
+            ...acc,
+            {
+              ingredient: value,
+              quantity: measure?.match(/[\s\d/\/]+/)?.join(''),
+              unit: measure?.match(/[a-z ]/gi)?.join(''),
+            },
+          ];
+        }
+        return acc;
+      }, []);
```

Comment:
1) all data changes you have to do in `normalizeRandomCocktailData` function 
2) you CAN'T mutate store - it's one of most important statement in redux. 
What it's mean: you can't change store fields directly in store like here you do - `state.details.ingredientsDetails =  ...`
If you want to change store, you need return completely new store object from reducer.

how it should be: 
1) `const cocktailData = normalizeRandomCocktailData(action.payload);` // it should be normalised and contain all you need based on payload: cocktailName, cocktailPicture, instructions, ingredientsDetails
2) you return new store from reducer:
return {
details: cocktailData
loading: false
}